src/nixos/machines/ignucius/releases/stable/ignucius-nixos-stable-install.sh

```diff
@@ -0,0 +1,148 @@
+
+	# Ensure that the RSD has the expected permissions
+	chown --verbose "root:root" "/run/agenix.d/1" # Ensure expected ownership
+	chmod --verbose 700 "/run/agenix.d/1" # Ensure expected permission
+
+	status "Ragenix Secret Directory has been set up"
+fi
+
+#! Set up the identity file
+status "Verifying the Identity File"
+
+[ -n "$ragenixIdentity" ] || ragenixIdentity="$HOME/.ssh/id_ed25519" # Try to use the default path
+
+# If the identity file is provided then use it to decrypt the secrets otherwise use hard-coded secrets
+if [ -f "$ragenixIdentity" ]; then
+	status "The identity file is provided trying to decrypt the secrets"
+
+	[ -s "/run/agenix/$machineName-disks-password" ] || age --identity "$ragenixIdentity" --decrypt --output "/run/agenix/$machineName-disks-password" "$secretPasswordPath"
+
+	[ -s "/run/agenix/$machineName-ssh-ed25519-private" ] || age --identity "$ragenixIdentity" --decrypt --output "/run/agenix/$machineName-ssh-ed25519-private" "$secretSSHHostKeyPath"
+
+	status "Decrypting of required secrets was successful"
+else
+	status "Required Identity File was not found, managing by using hard-coded secrets"
+
+	warn "BEWARE THAT USING HARD-CODED SECRETS IS A SECURITY HOLE!"
+
+	[ -s "/run/agenix/$machineName-disks-password" ] || echo "000000" > "/run/agenix/$machineName-disks-password"
+
+	[ -s "/run/agenix/$machineName-ssh-ed25519-private" ] || ssh-keygen -f "/run/agenix/$machineName-ssh-ed25519-private" -N ""
+fi
+
+#! Format the disks
+status "Formatting disks"
+disko \
+	--mode "disko" \
+	--root-mountpoint "/mnt" \
+	--debug \
+	--flake "$FLAKE_ROOT#$derivation"
+
+#! # Activate SWAP
+#! To prevent issues with lack of memory on systems with less than 16 GB of RAM
+# FIXME-QA(Krey): Do not run swapon if the swap is already activated
+# FIXME-QA(Kret): Use the device declaratively for activating swap
+# status "Activating swap"
+# swapon "$(realpath "$systemSwapDevice" || true)"
+# swapon "/dev/mapper/swap" || true
+
+# These have to be implemented for the installer to not fail with out of memory err
+# FIXME-QA(Krey): Do not run these if the size is already adjusted
+# mount -o remount,size=20G,noatime /nix/.rw-store
+# mount -o remount,size=5G,noatime /mnt
+
+#! Pre-build the system configuration
+status "Pre-building the system configuration"
+nixos-rebuild build --flake "$FLAKE_ROOT#$derivation" # pre-build the configuration
+
+#! Perform the installation
+status "Performing the system installation"
+nixos-install \
+	--verbose \
+	--root "/mnt" \
+	--flake "$FLAKE_ROOT#$derivation"
+
+#! # Insert the secret
+#! This is used to manage the chicken-and-an-egg problem with assigning system cryptographical keys
+status "Injecting cryptographical identification"
+age \
+	--identity "$ragenixIdentity" \
+	--decrypt \
+	--output "/mnt/nix/persist/system/etc/ssh/ssh_host_ed25519_key" \
+	"/run/agenix/$machineName-ssh-ed25519-private"
+
+#! Flash the Embedded Controller
+# FIXME(Krey)
```

Comment:
TBD

Reply:
The EC is not open-source only patchable so nothing here to do, for the bootloader we only flash the relevant region atm.. This might need better management in the future.